### DISCUSSION TITLE
Unkown field error in ModelSerializer

### DISCUSSION THREAD

**Discussion by aminiun** [encode/django-rest-framework#9013] [General]
Hi!

In `ModelSerializer` class, we have a function called `build_unknown_field` which raises error.
The error is something like: ```'Field name `invalid` is not valid for model `RegularFieldsModel`.'```

This is annoying sometimes for me, because I don't know where is the problem coming from, and I don't know which serializer is corrupt.

I was thinking to change the error to have the module path and the serializer class name (I have a PR to do that).

Is it OK to open an issue for that ?

**Comment by auvipy** [comment]
'Field name `invalid` is not valid for model `RegularFieldsModel`.' what code or name you are actually using?

**Reply by aminiun** [reply]
I took that exception from drf tests source code.

My real scenario is when I include some fields that does not exists in the model. For example assume that I have a model called `Store`, which does not have `token` field.

In this example, if I have a `ModelSerializer` like this:
```python
class StoreSerializer(ModelSerializer):
    ...
    class Meta:
        fields = (
            ...
            "token"
        )
```
Then I will face an exception like: ```'Field name `token` is not valid for model `Store`.'``` .

This can really be a headache, when I have multiple serializers for `Store` model. So I suggest to simply have the model path, to be more clear about which serializer is corrupt. Actually I have wrote the code already, I just want approval to make an issue and a PR.

**Reply by auvipy** [reply]
jjust share the PR. issue is not needed. don't forget to add appropriate regression tests :)
